src/main/java/com/pinterest/secor/parser/AvroMessageParser.java

```diff
@@ -0,0 +1,47 @@
+package com.pinterest.secor.parser;
+
+import com.google.common.primitives.Longs;
+import com.pinterest.secor.common.SecorConfig;
+import com.pinterest.secor.message.Message;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.BinaryDecoder;
+import org.apache.avro.io.DatumReader;
+import org.apache.avro.io.DecoderFactory;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Avro message parser extracts date partitions from avro messages.
+ * Uses: 'message.timestamp.name' to identify the field name.
+ */
+public class AvroMessageParser extends TimestampedMessageParser {
+    final DatumReader<GenericRecord> mDatumReader;
+    final String mFieldName;
+
+    public AvroMessageParser(SecorConfig config) throws IOException {
+        super(config);
+        Schema schema = new Schema.Parser().parse(new File(mConfig.getMessageInputAvroSchema()));
+        mFieldName = config.getMessageTimestampName();
+        mDatumReader = new GenericDatumReader<GenericRecord>(schema);
```

Comment:
If there are multiple topics being pulled in secor, you would need to different schemas for different topics, looks like this  config might need to be a map.